src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -505,6 +505,27 @@ class AvroSuite extends FunSuite with BeforeAndAfterAll {
     }
   }
 
+  test("correctly read long as date/timestamp type") {
+    TestUtils.withTempDir { tempDir =>
+      val sparkSession = spark
+      import sparkSession.implicits._
+
+      val currentTime = new Timestamp(System.currentTimeMillis())
+      val currentDate = new Date(System.currentTimeMillis())
+      val schema = StructType(Seq(
+        StructField("_1", DateType, false), StructField("_2", TimestampType, false)))
+      val writeDs = Seq((currentDate, currentTime)).toDS
+
+      val avroDir = tempDir + "/avro"
+      writeDs.write.avro(avroDir)
+      assert(spark.read.avro(avroDir).collect().length == 1)
+
+      val readDs = spark.read.schema(schema).avro(avroDir).as[(Date, Timestamp)]
```

Comment:
As can be seen from this test, this PR handles the case of reading Avro when Spark (SQL) schema is known. 

For the use case where schema is to be read from the Avro file(s) (the Avro schema), this test does not verify timestamp/date is being correctly deserialized.

Reply:
Yeah, this is for reading Avro with explicit Catalyst schema.

For the use case the schema is not given, we should also change the way we write the Avro file(s), IIUC.

I'd plan to try to add it in another PR. If you prefer, I can add the support here too.